src/Stryker.Core/Stryker.Core/Helpers/RangeHelper.cs

```diff
@@ -133,45 +134,51 @@ public static Position GetPosition(string text, int index)
             var line = 0;
             var col = 0;
 
-            for (var i = 0; i < System.Math.Min(index, text.Length); i++)
+            var reader = new StringReader(text);
+            string? latestLineContent = reader.ReadLine();
+            var currentIndex = 0;
+            while (latestLineContent != null)
             {
-                // TODO: handle x-platform
-                if (text[i] == '\n')
+                var endOfLineIndex = currentIndex + latestLineContent.Length;
+                var indexIsOnThisLine = index <= endOfLineIndex && index >= currentIndex;
+                if (indexIsOnThisLine)
                 {
-                    line++;
-                    col = 0;
+                    var indexOnThisLine = index - currentIndex;
+                    return PositionModule.mkPos(line, indexOnThisLine);
                 }
-                else
+
+                var lengthOfThisLine = latestLineContent.Length;
+                currentIndex += lengthOfThisLine;
+                latestLineContent = reader.ReadLine();
+                ++line;
+
+                if (latestLineContent == null)
```

Comment:
I guess this can be moved just after the loop?

Reply:
Yeah this felt a bit dodgy but I couldn't see a nicer way because of the overwriting of the `latestLineContent` I'll have another look at this later and see if I can tidy it up a bit more

Reply:
Thanks! Range arithmetic never really looks blazingly beautiful but we should try to do our best :D

Reply:
Hopefully that's at least more readable

Reply:
Yeah, this is cleaner. Thanks!